control/controlhttp/constants.go

```diff
@@ -18,3 +27,40 @@ const (
 	// to do the protocol switch is located.
 	serverUpgradePath = "/ts2021"
 )
+
+// DialOpts contains options when calling Dial.
+type DialOpts struct {
+	// The following fields are required
```

Comment:
typically required fields are positional parameters and Opts are for optional ones.

I wonder if this shouldn't just be like https://pkg.go.dev/net#Dialer and be a struct of things, rather than a `Dial(context.Context)` method.

Then there are no opts and it's less weird to have required "options".

Reply:
Pushed another commit at the end that does this; thoughts? It does remove a bunch of mostly-useless copying around, which is nice.